### PR TITLE
Introduced support of system keyring

### PR DESCRIPTION
**Pull request by catap**
Here a simple way that allows to migrate password to system keyrgin, and use it to run offlineimap.

### This PR

- [x] I've read the [DCO](http://www.offlineimap.org/doc/dco.html).
- [x] I've read the [Coding Guidelines](http://www.offlineimap.org/doc/CodingGuidelines.html)
- [x] The relevant informations about the changes stands in the commit message, not here in the message of the pull request.
- [x] Code changes follow the style of the files they change.
- [x] Code is tested (provide details).

### References

No references

### CODE REVIEW

File: offlineimap/init.py [hunk 1]
```diff
@@ -196,6 +196,14 @@ def __parse_cmd_options(self):
                           action="store_true", dest="mbnames_prune", default=False,
                           help="remove mbnames entries for accounts not in accounts")
 
+        parser.add_option("--ignore-keyring",
+                          action="store_true", dest="ignore_keyring", default=False,
+                          help="Ignore passwrod which is stored in system keyring")
```
Comment (rionda) [inline offlineimap/init.py:201]: Typo: passwrod -> password

Comment (catap) [reply offlineimap/init.py]: @rionda thanks!

**Comment by catap** [comment]
@thekix any chances to merge it?

**Comment by catap** [comment]
@chris001, @nicolas33 and @rionda any thoughts? I, personally, use it for a few years now on my macOS and it is part of MacPorts instalation of offline imap => probably is used not only by me.

**Comment by chris001** [comment]
@catap Yes I'm in favor. I'm reaching out to @nicolas33 and @thekix for their input.  @rionda your input would also be nice to have!

**Comment by catap** [comment]
@thekix I just rebased it to the last master. Any chance to have it reviewed as well?

**Comment by thekix** [comment]
Thanks @catap 

I will review this change tomorrow, I am a little busy, but I am on it.

Best regards,
kix

**Comment by catap** [comment]
Thanks to merge it!

**Comment by WhyNotHugo** [comment]
I wish this had been made optional; it adds an extra dependency which is not needed if using some other secrets management implementation.

**Comment by WhyNotHugo** [comment]
Er, actually, this introduces TWELVE dependencies, where offlineimap had a total of THREE.

I'd really appreciate if this feature could be made optional. Instead of unconditionally importing `keyring`, it could be imported only when explicitly required or when no fallback is available.

**Comment by WhyNotHugo** [comment]
Any thoughts on https://github.com/OfflineIMAP/offlineimap3/pull/191 to clean this up a bit?

**Comment by thekix** [comment]
Hi,

the keyring should be now optional. We are talking about changes in `requirements.txt`, `setup.py`and `pyproject.toml` at #182 

Regards,
kix

**Pull request merged**